src/Uno.UWP/ApplicationModel/DataTransfer/Clipboard.wasm.cs

```diff
@@ -57,6 +223,50 @@ private static void SetClipboardText(string text)
 			WebAssemblyRuntime.InvokeJS(command);
 		}
 
+		private static async Task<string> GetClipboardRichText(string mime, CancellationToken ct)
+		{
+			using (var data = await GetClipboardData(mime, ct))
+			{
+				return data?.ToString() ?? string.Empty;
+			}
+		}
+
+		private static Task<RandomAccessStreamReference> GetClipboardBitmap(string mime, CancellationToken ct)
+		{
+			return Task.FromResult(new RandomAccessStreamReference(async (ct) =>
+			{
+				MemoryStream memoryStream;
+
+				using (var data = await GetClipboardData(mime, ct))
+				{
+					memoryStream = new MemoryStream(data?.ToArray() ?? Array.Empty<byte>());
+				}
+
+				return new RandomAccessStreamOverStream(memoryStream).TrySetContentType();
+			}));
+		}
+
+		private static async Task<MarshalledData> GetClipboardData(string mime, CancellationToken ct)
+		{
+			var escapedMime = WebAssemblyRuntime.EscapeJs(mime);
+			var command = $@"
+				(async () => {{
+					var result = await {JsType}.getData(""{escapedMime}"");
+					if (!result) return null;
+					return JSON.stringify(result);
+				}})()
+				";
```

Comment:
This block should not be written in C# and use the "JsType" type in typecript instead.

Reply:
I wonder if such a block deserves its own TypeScript function, as the only purpose it serves is to marshal the JavaScript object to something C# can read.